pkg/controller/application/rolloutblock.go

```diff
@@ -0,0 +1,52 @@
+package application
+
+import (
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	apputil "github.com/bookingcom/shipper/pkg/util/application"
+	rolloutBlockOverride "github.com/bookingcom/shipper/pkg/util/rolloutblock"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (c *Controller) processRolloutBlocks(app *shipper.Application, nsRBs, gbRBs []*shipper.RolloutBlock) bool {
+	overrideRB, ok := app.GetAnnotations()[shipper.RolloutBlocksOverrideAnnotation]
+	if !ok {
+		overrideRB = ""
+	}
+	appOverrideRBs := rolloutBlockOverride.NewOverride(overrideRB)
+	rbs := append(nsRBs, gbRBs...)
+	existingRBs := rolloutBlockOverride.NewOverrideFromRolloutBlocks(rbs)
+	nonExistingRbs := appOverrideRBs.Diff(existingRBs)
```

Comment:
A comment on naming: a nonExistingRb makes me think it's up for creation, but from reading the code below it seems it's up for cleanup. My suggestion is to indicate that these items are outdated and are up for removal by renaming this var to something like: `obsoleteRbs` (or something around this sentiment).